src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_index_pattern/step_index_pattern.tsx

```diff
@@ -66,6 +60,10 @@ interface StepIndexPatternState {
 }
 
 export class StepIndexPattern extends Component<StepIndexPatternProps, StepIndexPatternState> {
+  static contextType = contextType;
+
+  declare readonly context: IndexPatternManagmentContextValue;
```

Comment:
Nit, maybe it is more future-proof to use `withKibana` higher order component instead of this legacy context API.

```js
export const StepIndexPattern = withKibana(StepIndexPatternPure);
```

Reply:
Or `useKibana` hook

```jsx
export const StepIndexPattern = props => {
  const {services} = useKibana();
  return <StepIndexPattern {...props} services={services} />;
};
```

Reply:
I think it's just a different way to pass the context. I found out that we use both ways in the application. Regarding `use Kibana` hook, I used it for functional components.